OptimizelySDK/OptimizelyJson.cs

```diff
@@ -0,0 +1,142 @@
+                Dict = (Dictionary<string, object>)ConvertIntoCollection(JObject.Parse(payload));
+                Payload = payload;
+            }
+            catch (Exception exception)
+            {
+                logger.Log(LogLevel.ERROR, "Provided string could not be converted to map.");
+                ErrorHandler.HandleError(new Exceptions.InvalidJsonException(exception.Message));
+            }
+        }
+        public OptimizelyJson(Dictionary<string, object> dict, IErrorHandler errorHandler, ILogger logger)
+        {
+            try
+            {
+                ErrorHandler = errorHandler;
+                Logger = logger;
+                Payload = JsonConvert.SerializeObject(dict);
+                Dict = dict;
+            }
+            catch (Exception exception)
+            {
+                logger.Log(LogLevel.ERROR, "Provided map could not be converted to string.");
+                ErrorHandler.HandleError(new Exceptions.InvalidJsonException(exception.Message));
+            }
+        }
+
+        override public string ToString()
+        {
+            return Payload;
+        }
+
+        public Dictionary<string, object> ToDictionary()
+        {
+            return Dict;
+        }
+
+        /// <summary>
+        /// If JSON Data is {"k1":true, "k2":{"k3":"v3"}}
+        ///
+        /// Set jsonPath to "k2" to access {"k3":"v3"} or set it to "k2.k3" to access "v3"
+        /// Set it to null or empty to access the entire JSON data but type must be Dictionary<string, object> as generic type.
+        /// </summary>
+        /// <param name="jsonPath">Key path for the value.</param>
+        /// <returns>Value if decoded successfully</returns>
+        public T GetValue<T>(string jsonPath)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(jsonPath))
+                {
+                    return (T)(object)Dict;
+                }
+                string[] path = jsonPath.Split('.');
+
+                Dictionary<string, object> currentObject = Dict;
+                for (int i = 0; i < path.Length - 1; i++)
+                {
+                    currentObject = currentObject[path[i]] as Dictionary<string, object>;
+                }
+                return GetObject<T>(currentObject[path[path.Length - 1]]);
+            }
+            catch (KeyNotFoundException exception)
+            {
+                Logger.Log(LogLevel.ERROR, "Value for JSON key not found.");
+                ErrorHandler.HandleError(new Exceptions.OptimizelyRuntimeException(exception.Message));
+            }
+            catch (Exception exception)
+            {
+                Logger.Log(LogLevel.ERROR, "Value for path could not be assigned to provided type.");
+                ErrorHandler.HandleError(new Exceptions.OptimizelyRuntimeException(exception.Message));
+            }
+            return default(T);
+        }
+
+        private T GetObject<T>(object o)
```

Comment:
this approach doesn't look good to me. Second thing, if we really need to check. use 

as instead of catching exception and doing nothing.
Revise this method.